### PR TITLE
[WIP] feat: bignumber integration

### PR DESCRIPTION
**Pull request by supaiku0**
## Proposed changes
This PR features:
- Bignum abstraction in `crypto/utils/bignum`
- Changes to the models for blocks, rounds, transactions and wallets to use Bignums for balances and stuff
- Adapted code to work with Bignums

It also solves #827.

The used Bignum implementation is [bigi](https://github.com/cryptocoinjs/bigi) since it is already being used by `crypto`. `bigi` (jsbn) is [slow](http://peterolson.github.io/BigInteger.js/benchmark/) , so it will be eventually completely replaced (together with `ecurve` which uses `bigi` as well...) by something much faster, but it's out of scope for this PR.

There's at least one [change](https://github.com/supaiku0/core/blob/bignumber-integration/packages/crypto/lib/models/block.js#L538) where I'm unsure about the consequences.

I'll rebase this PR onto https://github.com/ArkEcosystem/core/pull/969 once it's stable.

All tests that passed before are green and my node has been running without problems so far.

## Types of changes
- [X] Bugfix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [X] Refactoring (improve a current implementation without adding a new feature or fixing a bug)
- [X] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist
- [X] I have read the [CONTRIBUTING](https://docs.ark.io/developers/guidelines/contributing.html) documentation
- [X] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

**Comment by supaiku0** [comment]
Forgot to mention, but I made the following local changes to ark-js:
https://github.com/supaiku0/ark-js/commit/f3b9513d39f3372d8cb7d1f8ef434821f63b2fe7
```javascript
bb.writeLong(+transaction.amount.toString());
bb.writeLong(+transaction.fee.toString());
```
otherwise it will crash immediately.

**Comment by faustbrian** [comment]
@supaiku0 can you remove the sequelize files?

**Comment by supaiku0** [comment]
I will update this branch soon to drop [bigi](https://github.com/cryptocoinjs/bigi/) in favor of [bignumber.js](https://github.com/MikeMcl/bignumber.js/) and then it is ready to be merged.

**Comment by faustbrian** [comment]
@supaiku0 fix PR was merged so you can rebase and finish this.

**Comment by supaiku0** [comment]
@faustbrian ping

**Comment by supaiku0** [comment]
One more issue popped up, don't merge yet.

**Comment by faustbrian** [comment]
@supaiku0 Runs fine for me now, should be ready to merge?

File: packages/core-database-postgres/lib/spv.js [hunk 1]
```diff
@@ -108,9 +108,9 @@ module.exports = class SPV {
 
     for (const transaction of transactions) {
       let wallet = this.walletManager.findByPublicKey(transaction.senderPublicKey)
-      wallet.balance -= parseInt(transaction.amount) + parseInt(transaction.fee)
+      wallet.balance = wallet.balance.minus(transaction.amount).minus(transaction.fee)
 
-      if (wallet.balance < 0 && !this.isGenesis(wallet)) {
+      if (wallet.balance.toNumber() < 0 && !this.isGenesis(wallet)) {
```
Comment (faustbrian) [inline packages/core-database-postgres/lib/spv.js]: Could use `wallet.balance.isLessThan(0)`.

File: packages/core-database/lib/wallet-manager.js [hunk 2]
```diff
@@ -389,7 +392,7 @@ module.exports = class WalletManager {
    * @return {Boolean}
    */
   __canBePurged (wallet) {
-    return wallet.balance === 0 && !wallet.secondPublicKey && !wallet.multisignature && !wallet.username
+    return wallet.balance.toNumber() === 0 && !wallet.secondPublicKey && !wallet.multisignature && !wallet.username
```
Comment (faustbrian) [inline packages/core-database/lib/wallet-manager.js]: Could use `wallet.balance.isEqualTo(0)`.

**Pull request merged**